web/packages/teleport/src/Roles/RoleEditor/StandardEditor.test.tsx

```diff
@@ -77,3 +81,55 @@ const getAllSectionNames = () =>
 
 const getSectionByName = (name: string) =>
   screen.getByRole('heading', { level: 3, name }).closest('section');
+
+const TestServerAccessSpecsSection = ({
+  onChange,
+}: {
+  onChange(spec: ServerAccessSpec): void;
+}) => {
+  const [model, setModel] = useState<ServerAccessSpec>({
+    kind: 'node',
+    labels: [],
+    logins: [],
+  });
+  return (
+    <Validation>
+      <ServerAccessSpecSection
+        value={model}
+        onChange={spec => {
+          setModel(spec);
+          onChange(spec);
+        }}
+      />
+    </Validation>
+  );
+};
+
+test('editing server access specs', async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  render(<TestServerAccessSpecsSection onChange={onChange} />);
+  await user.click(screen.getByRole('button', { name: 'Add a Label' }));
+  await user.type(screen.getByPlaceholderText('label key'), 'some-key');
+  await user.type(screen.getByPlaceholderText('label value'), 'some-value');
+  await selectEvent.create(screen.getByLabelText('Logins'), 'root', {
+    createOptionText: 'Login: root',
+  });
+  await selectEvent.create(screen.getByLabelText('Logins'), 'some-user', {
+    createOptionText: 'Login: some-user',
+  });
+
+  expect(onChange).toHaveBeenLastCalledWith({
+    kind: 'node',
+    labels: [{ name: 'some-key', value: 'some-value' }],
+    logins: [newOptionFor('root'), newOptionFor('some-user')],
+  } as ServerAccessSpec);
+});
+
+const newOptionFor = (value: string) => ({
+  label: value,
+  value,
+  // This one is silently added by React Select to all newly created options.
+  // Surprise!
+  __isNew__: true,
```

Comment:
There's a chance that updating react-select will break this test. 
Instead of providing all fields, we can specify only the ones that are interesting for us:
https://jestjs.io/docs/expect#expectobjectcontainingobject

Reply:
That was actually my first approach, but it didn't work. I'll try again, maybe I did something wrong.